src/java.net.http/share/classes/jdk/internal/net/http/Stream.java

```diff
@@ -1472,35 +1488,48 @@ void completeResponseExceptionally(Throwable t) {
         @Override
         protected void handleResponse() {
             HttpHeaders responseHeaders = responseHeadersBuilder.build();
-            responseCode = (int)responseHeaders
-                .firstValueAsLong(":status")
-                .orElse(-1);
 
-            if (responseCode == -1) {
-                completeResponseExceptionally(new IOException("No status code"));
-            }
+            if (!finalPushResponseCodeReceived) {
+                responseCode = (int)responseHeaders
+                    .firstValueAsLong(":status")
+                    .orElse(-1);
 
-            this.response = new Response(
-                pushReq, exchange, responseHeaders, connection(),
-                responseCode, HttpClient.Version.HTTP_2);
+                if (responseCode == -1) {
+                    completeResponseExceptionally(new IOException("No status code"));
+                }
 
-            /* TODO: review if needs to be removed
-               the value is not used, but in case `content-length` doesn't parse
-               as long, there will be NumberFormatException. If left as is, make
-               sure code up the stack handles NFE correctly. */
-            responseHeaders.firstValueAsLong("content-length");
+                this.finalPushResponseCodeReceived = true;
```

Comment:
So the assumption here is that it doesn't make sense for a push promise response to have an intermediary 1xx code - so we don't support that and just return the first response we receive. I believe that's right. There may be a followup issue to log though, which would be to write test, and possibly revisit this code, to verify that we do fail in cases where we receive an unexpected sequence of frames, such as for instance, two final status codes, or several trailing header frames, or trailing headers followed by DataFrames (and this both on regular streams and pushed streams). However, this would be better handled in a followup PR. Maybe we could also investigate (in that followup) whether we should fail here if we receive a 1xx - instead of just returning it (on the other hand, maybe returning it and ignoring whatever follows is the right thing to do?) @Michael-Mc-Mahon what would you advise?